### ISSUE TITLE
Feature request: Panorama : show acknowledged alerts in a different colour in pie charts to inform someone is working on the issue and therefore is not a problem anymore

### ISSUE THREAD

**Issue by TomGudman** [sni/Thruk#493]
Hi,

We use panaroma and its pie charts to display our infrastructure status on panic boards (aka dashboards). The pie charts we also call "pacmans" are handy because we can see there is some sort of problem(s) and approximately the impact of things on a platform (hostgroup or regexp). 

However once a team start to work on an issue, they acknowledge the problem or eventually did put the services and/or hosts in maintenance mode. The fact action have been taken or work is currently done cannot be displayed on the dashboard because only the current state of the services or hosts are displayed (aka Warning or Critical state) whatever their ack, maintenance or notifications states are. Hence the idea to have another colour to state the fact that someone acknowledged the issue (e.g: blue). In the case of maintenance mode, either it is another colour or we could not take into consideration any host that is in maintenance mode in the pie chart - their state does not matter.

In short, can we add support like checkboxes in the pie chart attributes to display:
- Acknoledgements in another colour (e.g: blue)
- Maintenance state in another colour or simply do not account those services or host in the calculation of the overall pie chart status.

PS: I fully understand that this is not an easy change. I feel it requires consequent work.

Thoughts?

**Comment by sni** [comment]
Sounds good. If you want to implement this, let me know if i can help.

**Comment by TomGudman** [comment]
I can have a look at the problem and see if I can do it.
Can you give me a set of files to look into that are related to this matter?
That will help me to bootstrap me because otherwise I feel I may be quickly bogged in all these files / code ;-)

**Comment by sni** [comment]
Sure, when talking about the pie widgets, then the js code is here:
https://github.com/sni/Thruk/blob/master/plugins/plugins-available/panorama/root/js/panorama_js_panlet_piechart_stats.js
(using the class from here: https://github.com/sni/Thruk/blob/master/plugins/plugins-available/panorama/root/js/panorama_js_panlet_piechart.js)

The controller is here: https://github.com/sni/Thruk/blob/master/plugins/plugins-available/panorama/lib/Thruk/Controller/panorama.pm#L1573

When talking about the pie icons, then everything is in here:
https://github.com/sni/Thruk/blob/master/plugins/plugins-available/panorama/root/js/panorama_js_panlet_icon_widgets.js
and
https://github.com/sni/Thruk/blob/master/plugins/plugins-available/panorama/root/js/panorama_js_panlet_icon_widgets_settings.js
